lib/sdam/server_selectors.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+const ServerType = require('./server_description').ServerType;
+const TopologyType = require('./topology_description').TopologyType;
+const ReadPreference = require('../topologies/read_preference');
+const MongoError = require('../error').MongoError;
+
+// max staleness constants
+const IDLE_WRITE_PERIOD = 10000;
+const SMALLEST_MAX_STALENESS_SECONDS = 90;
+
+function writableServerSelector() {
```

Comment:
Is there a reason that we are returning a function here? There doesn't appear to be any closure at work here.

Reply:
to keep it consistent with the `readPreferenceServerSelector` below. The idea is to eventually be able to support arbitrary user-provided selectors so I wanted to standardize on a form.